contracts/tokens/RewardEthToken.sol

```diff
@@ -135,14 +147,27 @@ contract RewardEthToken is IRewardEthToken, OwnablePausableUpgradeable, ERC20Per
      * @dev See {IRewardEthToken-updateRewardCheckpoint}.
      */
     function updateRewardCheckpoint(address account) external override {
-        checkpoints[account] = Checkpoint(rewardPerToken, balanceOf(account));
+        Checkpoint memory checkpoint = Checkpoint(balanceOf(account).toUint128(), rewardPerToken);
+        checkpoints[account] = checkpoint;
```

Comment:
Why did you introduce the local variable here?   From the gas usage perspective, this spends more gas than directly updating the storage without the local variable.

Reply:
Reverted to the previous version.